.github/workflows/CD.yml

```diff
@@ -629,16 +629,34 @@ jobs:
           src: public.ecr.aws/${{ env.ECR_REPO }}:${{ needs.release-checking.outputs.version }}
           dst: ${{ env.IMAGE_NAMESPACE }}/${{ env.IMAGE_NAME }}:${{ needs.release-checking.outputs.version }}
 
+  mirror_adot_operator:
+    runs-on: ubuntu-latest
+    needs: release-latest-image
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.17.7
+
+      - name: Compare version with Dockerhub latest
+        id: version
+        if: needs.release-latest-image.outputs.cache-hit != 'true'
+        run: |
+          TAG="${{ needs.release-checking.outputs.version }}"
+          TARGET_VERSION=$TAG bash tools/workflow/docker-version-compare.sh
+
       - name: Configure AWS Credentials
-        if: steps.release-latest-image.outputs.cache-hit != 'true'
+        if: needs.release-latest-image.outputs.cache-hit != 'true'
         uses: aws-actions/configure-aws-credentials@v1
         with:
           aws-access-key-id: ${{ secrets.RELEASE_KEY_ID }}
           aws-secret-access-key: ${{ secrets.RELEASE_SECRET }}
           aws-region: us-west-2
 
       - name: Mirror ADOT operator
-        if: ${{ github.event.inputs.skip-adot-operator-mirroring == 'false' && (steps.release-latest-image.outputs.cache-hit != 'true' && (steps.version.outputs.major-update == 'true' || steps.version.outputs.minor-update == 'true') || steps.version.outputs.same-version == 'true') }}
+        if: ${{ github.event.inputs.skip-adot-operator-mirroring == 'false' && (needs.release-latest-image.outputs.cache-hit != 'true' && (steps.version.outputs.major-update == 'true' || steps.version.outputs.minor-update == 'true') || steps.version.outputs.same-version == 'true') }}
```

Comment:
Same issue with this cache hit. I know you did not write this but can you explain the conditional logic behind the version comparison? Should the operator be bound to only be mirrored during major/minor or same collector version CD runs? Why do we not do it on patch releases? 

This may have been answered in the previous PR but is running the mirror script when ADOT Operator is the same version as Upstream Operator. The `docker-version-compare` compares the collector versions and I am wondering how that is related to operator versions. 

I suspect this could be tech debt and we should fully understand the reasoning for it before moving forward.

Reply:
I don't understand the conditional logic either, I think it would make sense for the operator to be mirrored on major/minor/patch releases and NOT same Collector version CD runs. The `docker-version-compare` DOES compare collector versions, but if the idea is that if the ADOT Operator releases are still tied to ADOT Collector releases (+ the upstream OTEL Operator.Collector are in sync), then a check for collector version is still the same. Either way, when the ADOT Operator is already the same version as the upstream, no image ends up being mirrored in the end regardless

Reply:
Just changed the `same-version` check to a `patch-update` check as well